tests/perf/profile_w1r3.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Workload W1R3 profiling script. This is not an officially supported Google product."""
+
+import logging
+import os
+import random
+import time
+import uuid
+
+from functools import partial, update_wrapper
+
+from google.cloud import storage
+
+import _perf_utils as _pu
+
+
+def WRITE(bucket, blob_name, checksum, size, args, **kwargs):
+    """Perform an upload and return latency."""
+    blob = bucket.blob(blob_name)
+    file_path = f"{os.getcwd()}/{uuid.uuid4().hex}"
+    # Create random file locally on disk
+    with open(file_path, "wb") as file_obj:
+        file_obj.write(os.urandom(size))
+
+    start_time = time.monotonic_ns()
```

Comment:
Usually I just use time.time(), is that not sufficient for these cases?

Reply:
Mainly using this in performance profiling for accuracy. The monotonic clock is not affected by system clock updates, and `time.monotonic_ns()` avoids the precision loss caused by the `time.time()` float type